ALEC-228: Create Quick Start section

emarsh1995 opened the pull request:
- Add `quick-start/data-permissions.png`, `/engine-config.png`, `/situation-details.png`, `/situation-list.png`, `/situation-timeline.png`
- Add `quick-start/engine-config.adoc`, `/introduction.adoc`, `/situations.adoc`
- Change `engines/cluster.adoc` -> `engines/introduction.adoc` & update
- Change `engines/dbscan.adoc` -> `engines/clustering.adoc` & update
- Update `engines/deeplearning.adoc`
- Update `ROOT/nav.adoc`
- Update `about/welcome.adoc`

### External References
- JIRA (Issue Tracker): https://opennms.atlassian.net/browse/ALEC-228

Commit:
ALEC-228: How to Use section

- Add quick-start/data-permissions.png, /engine-config.png
- Add quick-start/engine.adoc, /introduction.adoc, /situations.adoc

Commit:
ALEC-228: How to Use section

- Update engines/cluster.adoc
- Update quick-start/engine.adoc

Commit:
ALEC-228: How to Use section

- Update engines/dbscan.adoc

Commit:
ALEC-228: How to Use section

- Update engines/cluster.adoc, /dbscan.adoc, /deeplearning.adoc
- Add quick-start/situation-details.png, /situation-list.png, /situation-timeline.png
- Update quick-start/engine.adoc, /situations.adoc
- Update ROOT/nav.adoc

Commit:
ALEC-228: How to Use section

- Change engines/cluster.adoc -> engines/introduction.adoc
- Change engines/dbscan.adoc -> engines/clustering.adoc
- Change quick-start/engine.adoc -> quick-start/engine-config.adoc
- Update quick-start/situations.adoc
- Update ROOT/nav.adoc

Commit:
Merge branch 'develop' into jira/alec-228

Commit:
ALEC-228: How to Use section

- Update about/welcome.adoc
- Update quick-start/situations.adoc

Bonrob2 commented on docs/modules/quick-start/pages/engine-config.adoc:
```diff
@@ -0,0 +1,41 @@
+= Step 1: First-Time Login and Engine Configuration
+
+This section explains first-time login tasks and how to configure a correlation engine.
+ALEC includes two correlation engines: Clustering and Deep Learning.
```
```suggestion
ALEC includes two correlation engines: clustering and deep learning.
```

Bonrob2 commented on docs/modules/quick-start/pages/engine-config.adoc:
```diff
@@ -0,0 +1,41 @@
+
+This section explains first-time login tasks and how to configure a correlation engine.
+ALEC includes two correlation engines: Clustering and Deep Learning.
+The engine that you choose determines how alerts are grouped into clusters.
```
```suggestion
The engine that you choose determines how ALEC groups alerts.
```
Is it important they are grouped into clusters, or just that they are grouped? Does "cluster" mean the same as "situation" or is there a distinction we need to make here?

emarsh1995 replied:
Clusters, from what I understand, are the prerequisite for situations. Clusters simply group alarms together; situations represent a synthetic "parent" alarm (as listed in the glossary). I'll get some clarification before we go forward.

Bonrob2 commented on docs/modules/quick-start/pages/engine-config.adoc:
```diff
@@ -0,0 +1,41 @@
+This section explains first-time login tasks and how to configure a correlation engine.
+ALEC includes two correlation engines: Clustering and Deep Learning.
+The engine that you choose determines how alerts are grouped into clusters.
+This Quick Start guide explains how to set up and configure the xref:engines:clustering.adoc[Clustering] engine.
```
```suggestion
This Quick Start guide explains how to set up and configure the xref:engines:clustering.adoc[clustering] engine.
```

Bonrob2 commented on docs/modules/quick-start/pages/engine-config.adoc:
```diff
@@ -0,0 +1,41 @@
+ALEC includes two correlation engines: Clustering and Deep Learning.
+The engine that you choose determines how alerts are grouped into clusters.
+This Quick Start guide explains how to set up and configure the xref:engines:clustering.adoc[Clustering] engine.
+For information on the Deep Learning engine, see xref:engines:deeplearning.adoc[Deep Learning].
```
```suggestion
The xref:engines:deeplearning.adoc[deep learning] engine requires additional setup, and is beyond the scope of this Quick Start.
```
Something like this to indicate why we aren't covering it here.
We should confirm if anything happens if they do select deep learning and they haven't installed the dl engine. Do they get an error message? Should this even be an option if they don't have it installed?

Bonrob2 commented on docs/modules/quick-start/pages/engine-config.adoc:
```diff
@@ -0,0 +1,41 @@
+Select *Yes* or *No*, and click *Continue*.
+The correlation engine configuration screen is displayed.
+
+== Configure Clustering engine
```
```suggestion
== Configure clustering engine
```

Bonrob2 commented on docs/modules/quick-start/pages/engine-config.adoc:
```diff
@@ -0,0 +1,41 @@
+image::engine-config.png[Correlation engine selection prompt; includes details about the Clustering and Deep Learning correlation engines, 900]
+
+Choose *Clustering*, and click *Continue*.
+The Clustering engine is now set.
```
```suggestion
The clustering engine is now set.
```

Bonrob2 commented on docs/modules/quick-start/pages/engine-config.adoc:
```diff
@@ -0,0 +1,41 @@
+Choose *Clustering*, and click *Continue*.
+The Clustering engine is now set.
+
+NOTE: After you select a correlation engine, it can be changed and updated only via the Karaf shell.
```
```suggestion
NOTE: After you select a correlation engine, you can change and update it only via the Karaf shell.
```

Bonrob2 commented on docs/modules/quick-start/pages/introduction.adoc:
```diff
@@ -0,0 +1,6 @@
+
+= Quick Start
+
+This Quick Start guide assumes that you have already https://docs.opennms.com/horizon/31/operation/quick-start/inventory.html[deployed an OpenNMS instance], followed the https://docs.opennms.com/horizon/31/operation/quick-start/introduction.html[core Quick Start guide], and xref:install:common_install.adoc[installed ALEC].
```
Will have to update this once we've merged ALEC-227.

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+:imagesdir: ../assets/images
+= Step 2: View and Create Situations
+
+Now that you have xref:engine-config.adoc[selected the Clustering engine], ALEC automatically begins to identify situations.
```
```suggestion
Now that you have xref:engine-config.adoc[selected the clustering engine], ALEC automatically begins to identify situations.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+
+== View situations
+
+After alarms have been triggered in OpenNMS, they are evaluated by the Clustering engine and grouped into clusters as appropriate.
```
```suggestion
After alarms have been triggered in OpenNMS, the clustering engine evaluates and groups them into situations as appropriate.
```
As per earlier question. Is there a distinction between a cluster and a situation?

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+== View situations
+
+After alarms have been triggered in OpenNMS, they are evaluated by the Clustering engine and grouped into clusters as appropriate.
+You can view current situations in the OpenNMS web UI: click *Plugins > ALEC* in the top menu bar.
```
```suggestion
To view current situations in the OpenNMS web UI, click *Plugins > ALEC* in the top menu bar.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+image::situation-list.png[ALEC Situation List page displaying active situations, 800]
+
+Active situations are sorted from most to least severe.
+You can modify the sorting method by using the filters at the left of the screen.
```
```suggestion
Use the filters at the left of the screen to modify the sorting method.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+== Create a situation
+
+You may occasionally notice alarms that are related, but that were not grouped into a situation.
+In these cases, you can create a situation to represent such relationships.
```
```suggestion
In these cases, you can create a situation to represent such relationships:
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+
+You may occasionally notice alarms that are related, but that were not grouped into a situation.
+In these cases, you can create a situation to represent such relationships.
+
+Follow these steps to create a situation:
```
```suggestion

```

emarsh1995 replied:
```suggestion
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+Now that you have xref:engine-config.adoc[selected the Clustering engine], ALEC automatically begins to identify situations.
+This requires new alarms to be triggered within your OpenNMS environment, so--depending on your network's usual activity--this may take some time.
+
+== View situations
```
```suggestion
== View all situations
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+You can modify the sorting method by using the filters at the left of the screen.
+To return to the default sorting and filtering settings, click *Reset Filters*.
+
+To view more information on a specific situation, click its entry in the list.
```
```suggestion
== View a specific situation

To view more information on a specific situation, click its entry in the list.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+
+Now that you have xref:engine-config.adoc[selected the Clustering engine], ALEC automatically begins to identify situations.
+This requires new alarms to be triggered within your OpenNMS environment, so--depending on your network's usual activity--this may take some time.
+
```
We should add a short description of what the recommended user process/workflow is. Something that mentions that as they see situations, they will want to evaluate and address them. Beyond the basic monitoring through situations, they can also train the UI, which is a more advanced topic.

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+image::situation-details.png[Situation details page for a critical situation that affects two nodes, 800]
+
+From this page, you can acknowledge the situation (indicate that it has been seen and is being addressed), clear it (mark the situation as not relevant), or reject it (indicate that the clustered alarms are not related).
+You can also take action on individual alarms within the cluster by selecting them in the *Alarms* list.
```
Check with the devs, but I wonder if it makes more sense to address alarms first, and then clear situations.

This section might work better with a workflow like: **view all situations**, **view a specific situation**, **manage alarms** (include how to clear, acknowledge, escalate them, and the memos/sticky notes), **manage the situation**, **view the situation timeline**.

Suggest adding separate subheadings for these items so the information is easier to find. In some cases (like the memos) this could be stepped out into a procedure, but starting from within the screen. For example:
To create a memo:
1. Click the pencil symbol beside the type of memo ...
2. Type your note.
3. Click the green check symbol to save it.

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,51 @@
+From this page, you can acknowledge the situation (indicate that it has been seen and is being addressed), clear it (mark the situation as not relevant), or reject it (indicate that the clustered alarms are not related).
+You can also take action on individual alarms within the cluster by selecting them in the *Alarms* list.
+
+You can create memos for both the situation overall and for individual alarms.
```
What is the difference between a sticky memo and a journal memo?

emarsh1995 replied:
I'll clarify this as well.

Bonrob2 reviewed (changes requested):
Some suggestions around restructuring it for better workflow. Thanks!

Bonrob2 commented:
We should update our doc outline to include these topics, or add them to the QS:

- Move alarms to another situation
- View unassociated alarms

Bonrob2 commented on docs/modules/engines/pages/clustering.adoc:
```diff
@@ -0,0 +1,19 @@
+:imagesdir: ../assets/images
+= Clustering Engine
+
+The Clustering engine is a xref:introduction.adoc[correlation engine] that builds alarm clusters using the https://en.wikipedia.org/wiki/DBSCAN[DBSCAN algorithm].
```
```suggestion
The clustering engine is a xref:introduction.adoc[correlation engine] that uses the https://en.wikipedia.org/wiki/DBSCAN[DBSCAN algorithm] to build alarm clusters.
```

Bonrob2 commented on docs/modules/engines/pages/clustering.adoc:
```diff
@@ -0,0 +1,19 @@
+= Clustering Engine
+
+The Clustering engine is a xref:introduction.adoc[correlation engine] that builds alarm clusters using the https://en.wikipedia.org/wiki/DBSCAN[DBSCAN algorithm].
+It draws information from the locally-persisted network inventory graph; this graph is automatically created and maintained by ALEC, and is based upon information from the associated xref:datasources:overview.adoc[datasource].
```
```suggestion
It draws information from the locally persisted network inventory graph.
ALEC automatically creates and maintains this graph, which is based on information from the associated xref:datasources:overview.adoc[datasource].
```

Bonrob2 commented on docs/modules/engines/pages/clustering.adoc:
```diff
@@ -0,0 +1,19 @@
+
+== DBSCAN algorithm
+
+The DBSCAN algorithm that forms the basis for the Clustering engine measures the distance between alarms on the network inventory graph to determine whether they should be correlated into a cluster.
```
```suggestion
The DBSCAN algorithm that forms the basis for the clustering engine measures the distance between alarms on the network inventory graph to determine whether they should be correlated into a cluster.
```

Bonrob2 commented on docs/modules/engines/pages/clustering.adoc:
```diff
@@ -0,0 +1,19 @@
+
+== Performance
+
+The Clustering engine performs well when fewer than 500 candidate alarms exist.
```
```suggestion
The clustering engine performs well when fewer than 500 candidate alarms exist.
```

Bonrob2 replied:
What happens when there are more than 500? Is it possible that users would ever have that many alarms? If they already have over 500 alarms when they install ALEC, what happens? Is it just super slow?

Bonrob2 commented on docs/modules/engines/pages/clustering.adoc:
```diff
@@ -0,0 +1,19 @@
+== Performance
+
+The Clustering engine performs well when fewer than 500 candidate alarms exist.
+Note that alarms are considered to be candidates for correlation only when they have been created or updated within the last two hours (this interval can be configured).
```
```suggestion
Note that ALEC considers only recent alarms (created or updated within the previous two hours) as candidates for correlation.
You can configure this interval.
```
We'll have to add a link to this procedure on how to configure the interval. I'll note that it should go in the Maintenance section I'm creating as part of the installation work.

Bonrob2 commented on docs/modules/engines/pages/clustering.adoc:
```diff
@@ -0,0 +1,19 @@
+The Clustering engine performs well when fewer than 500 candidate alarms exist.
+Note that alarms are considered to be candidates for correlation only when they have been created or updated within the last two hours (this interval can be configured).
+
+The engine can be used on monitored networks that periodically experience more than 500 active alarms, since many of them will age out over time.
```
```suggestion
You can use the engine on monitored networks that periodically experience more than 500 active alarms, since many of them will age out over time.
```
I would move this up after 15, but it also seems to contradict it a bit.

Bonrob2 commented on docs/modules/engines/pages/clustering.adoc:
```diff
@@ -0,0 +1,19 @@
+Note that alarms are considered to be candidates for correlation only when they have been created or updated within the last two hours (this interval can be configured).
+
+The engine can be used on monitored networks that periodically experience more than 500 active alarms, since many of them will age out over time.
+The ALEC plugin's performance, however, may degrade when more than 500 active candidate alarms exist in your OpenNMS instance.
```
Ah, it's here! Please move up to under line 15 and combine.

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
```diff
@@ -1,89 +1,85 @@
+= Deep Learning Engine
 
-image::ludwig_training.png[Ludwing training example,600]
+The Deep Learning engine is a xref:introduction.adoc[correlation engine] that uses a https://www.tensorflow.org/[TensorFlow] model to build alarm clusters.
```
```suggestion
The deep learning engine is a xref:introduction.adoc[correlation engine] that uses a https://www.tensorflow.org/[TensorFlow] model to build alarm clusters.
```

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
```diff
@@ -1,89 +1,85 @@
 
-image::ludwig_training.png[Ludwing training example,600]
+The Deep Learning engine is a xref:introduction.adoc[correlation engine] that uses a https://www.tensorflow.org/[TensorFlow] model to build alarm clusters.
+It draws information from the locally-persisted network inventory graph; this graph is automatically created and maintained by ALEC, and is based upon information from the associated xref:datasources:overview.adoc[datasource].
```
```suggestion
It draws information from the locally persisted network inventory graph.
Alec automatically creates and maintains this graph, which is based on information from the associated xref:datasources:overview.adoc[datasource].
```

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
```diff
@@ -1,89 +1,85 @@
-* Difference in time (numerical)
-* Distance on the graph (numerical)
+To create clusters using TensorFlow, ALEC first reduces the clustering calculation to a binary classification that asks, "Are these two alarms related?"
+When the Deep Learning engine determines that two or more alarms are related, they are added to a cluster.
```
```suggestion
When the deep learning engine determines that two or more alarms are related, they are added to a cluster.
```

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
```diff
@@ -1,89 +1,85 @@
+== Ludwig
 
-== Using the engine
+The current model definition for the Deep Learning engine has been developed using https://github.com/uber/ludwig[Ludwig]:
```
```suggestion
The current model definition for the deep learning engine has been developed using https://github.com/uber/ludwig[Ludwig]:
```

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
```diff
@@ -1,89 +1,85 @@
 
-To use the engine, install the `alec-engine-deeplearning` feature.
+.Ludwig training example
+image::ludwig_training.png[Shell interface showing an example of training input and output for Ludwig, 600]
```
```suggestion
image::ludwig_training.png[Shell interface that shows an example of training input and output for Ludwig, 600]
```

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
```diff
@@ -1,89 +1,85 @@
+== Train the engine
 
-TIP: Once installed add the feature to the `alec.boot` in a new line with `alec-engine-deeplearning wait-for-kar=opennms-alec-plugin` to ensure that engine is re-installed when the services are restarted.
+The Deep Learning engine must be trained via the Karaf shell before it can be used to correlate alarms.
```
```suggestion
The deep learning engine must be trained via the Karaf shell before ALEC can use it to correlate alarms.
```

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
```diff
@@ -1,89 +1,85 @@
+The Deep Learning engine must be trained via the Karaf shell before it can be used to correlate alarms.
 
-== Training
+WARNING: These instructions are included to help you get started with the engine.
```
```suggestion
IMPORTANT: These instructions are included to help you get started with the engine.
```

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
```diff
@@ -1,89 +1,85 @@
+=== Export the model
 
-Adapt the following script to export the model to a format that can be used in ALEC:
+Use the following script to export the trained model to a format that can be used in ALEC:
```
```suggestion
Use the following script to export the trained model to a format that ALEC can use:
```

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
````diff
@@ -108,38 +104,45 @@ chmod +x export_model.py
+NOTE: If the command results are negative, you must retrain and re-export the training model.
 
-```
+If the command results are positive, you can then configure the Deep Learning engine to use the model:
````
```suggestion
If the command results are positive, you can then configure the deep learning engine to use the model:
```

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
````diff
@@ -1,89 +1,85 @@
-```
 
-=== Vectorize the datasets
+=== Vectorize datasets
````
dataset ... one or two words? We use two words below.

emarsh1995 replied:
I think we're leaning towards one word, but I initially started having it as two; I must've missed an instance below. Thanks for catching it!

Bonrob2 replied:
TBH, I always forget. It's one I have to look up almost every time.

Bonrob2 commented on docs/modules/engines/pages/deeplearning.adoc:
```diff
@@ -108,38 +104,45 @@ chmod +x export_model.py
+You can run simulations to verify that the training model clusters alerts as expected.
 
-Generate situations:
+First, use the following commands to generate situations based on the data set snapshot from earlier:
```
```suggestion
First, use the following commands to generate situations based on the dataset snapshot from earlier:
```
dataset?

Bonrob2 commented on docs/modules/engines/pages/introduction.adoc:
```diff
@@ -0,0 +1,21 @@
+= Correlation Engines
+
+Correlation engines are the underlying processes that drive ALEC.
+Two engines are implemented by default: xref:clustering.adoc[Clustering] and xref:deeplearning.adoc[Deep Learning].
```
```suggestion
Two engines are implemented by default: xref:clustering.adoc[clustering] and xref:deeplearning.adoc[deep learning].
```

Bonrob2 commented on docs/modules/engines/pages/introduction.adoc:
```diff
@@ -0,0 +1,21 @@
+
+Correlation engines are the underlying processes that drive ALEC.
+Two engines are implemented by default: xref:clustering.adoc[Clustering] and xref:deeplearning.adoc[Deep Learning].
+Both extend the cluster engine, which builds network topology graphs based on information provided by your datasource (see xref:datasources:overview.adoc[ALEC Datasources]).
```
```suggestion
Both extend the correlation engine, which builds network topology graphs based on information provided by your datasource (see xref:datasources:overview.adoc[ALEC Datasources]).
```

emarsh1995 replied:
We may want to talk to the team about the naming here, since the underlying cluster engine is different from the cluster*ing* engine. I'll note it for tomorrow.

Bonrob2 commented on docs/modules/engines/pages/introduction.adoc:
```diff
@@ -0,0 +1,21 @@
+Two engines are implemented by default: xref:clustering.adoc[Clustering] and xref:deeplearning.adoc[Deep Learning].
+Both extend the cluster engine, which builds network topology graphs based on information provided by your datasource (see xref:datasources:overview.adoc[ALEC Datasources]).
+
+== Cluster engine
```
correlation engine?

Bonrob2 commented on docs/modules/engines/pages/introduction.adoc:
```diff
@@ -0,0 +1,21 @@
+
+== Cluster engine
+
+The cluster correlation engine builds a network topology graph.
```
```suggestion
The correlation engine builds a network topology graph.
```

Bonrob2 commented on docs/modules/engines/pages/introduction.adoc:
```diff
@@ -0,0 +1,21 @@
+== Cluster engine
+
+The cluster correlation engine builds a network topology graph.
+The graph is based on network inventory information provided by the datasource, and is persisted in local memory.
```
```suggestion
The graph is based on network inventory information the datasource provides, and is persisted in local memory.
```

Bonrob2 commented on docs/modules/engines/pages/introduction.adoc:
```diff
@@ -0,0 +1,21 @@
+Inventory objects are added to the graph as vertices, and relationships among them are added as edges.
+Alarms are then attached to the vertex of the inventory object that they reference.
+
+On every refresh interval (which defaults to 30 seconds), ALEC runs a clustering algorithm against all alarms on the graph.
```
I assume this interval is set in the OpenNMS core, so probably no need to document, but let's verify.

Bonrob2 commented on docs/modules/engines/pages/introduction.adoc:
```diff
@@ -0,0 +1,21 @@
+This process outputs a list of alarm clusters.
+The engine then maps clusters to situations, and sends that situation data back to the datasource.
+
+The underlying cluster engine is a generic implementation.
```
```suggestion
The underlying correlation engine is a generic implementation.
```

Bonrob2 reviewed (changes requested):
I seem to have missed a few files.

emarsh1995 committed:
Apply suggestions from code review

Co-authored-by: Bonrob2 <59453630+Bonrob2@users.noreply.github.com>

Commit:
ALEC-228: Quick Start section

- Update engines/clustering.adoc, /introduction.adoc
- Update quick-start/engine-config.adoc, /situations.adoc

Bonrob2 commented on docs/modules/quick-start/pages/engine-config.adoc:
```diff
@@ -0,0 +1,42 @@
+The engine that you choose determines how ALEC groups alerts into clusters.
+
+This Quick Start guide explains how to set up and configure the xref:engines:clustering.adoc[clustering] engine.
+The xref:engines:deeplearning.adoc[deep learning] engine requires additional setup, and is beyond the scope of this Quick Start guide.
```
```suggestion
The xref:engines:deeplearning.adoc[deep learning] engine requires additional setup, and is covered as an advanced topic in the administration section.
```
Walking back my previous edit, as we have Quick Start twice in the same paragraph. 
We can add a link to the topic when we merge the installation PR.

Bonrob2 commented on docs/modules/quick-start/pages/engine-config.adoc:
```diff
@@ -0,0 +1,42 @@
+
+== First-time login and data choices
+
+As part of the introduction to ALEC, OpenNMS prompts for permission to send ALEC usage data to The OpenNMS Group.
```
```suggestion
When you first log in and navigate to the ALEC plugin, OpenNMS prompts for permission to send usage data to The OpenNMS Group.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,97 @@
+Now that you have xref:engine-config.adoc[selected the clustering engine], ALEC automatically begins to identify situations.
+This requires new alarms to be triggered within your OpenNMS environment, so--depending on your network's usual activity--this may take some time.
+
+As alarms are grouped into clusters, situations are created to represent their relationships.
```
```suggestion
As alarms are grouped into clusters, ALEC creates situations to represent their relationships.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,97 @@
+This requires new alarms to be triggered within your OpenNMS environment, so--depending on your network's usual activity--this may take some time.
+
+As alarms are grouped into clusters, situations are created to represent their relationships.
+It is recommended that you evaluate situations as you see them, and address them as appropriate.
```
```suggestion
We recommend that you evaluate situations as you see them, and address them as appropriate.
```
This is great. These two sentences are really helpful for the user.

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,97 @@
+* *Reject:* Indicate that the alarms in the situation are not related.
+
+You can also maintain sticky memos and journal memos for the situation.
+Sticky memos are used to note key information for the situation overall (for example, the potential root cause).
```
```suggestion
Sticky memos note key information for the situation overall (for example, the potential root cause).
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,97 @@
+
+You can also maintain sticky memos and journal memos for the situation.
+Sticky memos are used to note key information for the situation overall (for example, the potential root cause).
+Journal memos are used to log any important work or investigation steps; this can include root cause analysis or developing a fix for the underlying issue.
```
```suggestion
Journal memos log any important work or investigation steps; this can include root cause analysis or developing a fix for the underlying issue.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,97 @@
+
+. Navigate to a situation's details page.
+. Click the *pencil* symbol beside *Sticky Memo* or *Journal Memo*, as desired.
+. Type your note in the text box that is displayed.
```
```suggestion
. Type your note in the text box.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,97 @@
+* *Acknowledge:* Indicate that the alarm has been seen and is being addressed.
+Your account will take ownership of the alarm within OpenNMS.
+* *Move:* Move the alarm to a different situation.
+This should be used when another situation more accurately encompasses an issue.
```
```suggestion
Use this when another situation more accurately encompasses an issue.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,97 @@
+* *Move:* Move the alarm to a different situation.
+This should be used when another situation more accurately encompasses an issue.
+* *Remove:* Remove the alarm from the situation.
+This should be used when an alarm has been incorrectly associated with a situation.
```
```suggestion
Use this when an alarm has been incorrectly associated with a situation.
```

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,97 @@
+This should be used when another situation more accurately encompasses an issue.
+* *Remove:* Remove the alarm from the situation.
+This should be used when an alarm has been incorrectly associated with a situation.
+
```
```suggestion
NOTE: The move and remove options help train ALEC to better identify relationships between alarms and more accurately cluster them into situations.

```
(If this is indeed correct.)

Bonrob2 commented on docs/modules/quick-start/pages/situations.adoc:
```diff
@@ -0,0 +1,97 @@
+
+You may occasionally notice alarms that are related, but that were not grouped into a situation.
+In these cases, you can create a situation to represent such relationships:
+
```
```suggestion
NOTE: You can also view and move unassociated alarms into existing situations.
(See xref: .... [Manage unassociated alarms].)
```
Suggest adding a topic below this "Manage unassociated alarms" that describes the process of clicking the "View Unassociated Alarms" button and then moving them to a situation. We have described the "move" feature as part of alarm management above, but it would be good to show them this route that is specific to unassociated alarms. We also need to check with Ben or Gerardo if unassociated alarms *must* be in a situation in order to manage them. I don't see a way to clear, acknowledge, or reject them from that screen.

Bonrob2 reviewed (changes requested):
Very nice work! The Quick Start flows very well. A few nits, and something we need to clarify regarding "unassociated alarms".

emarsh1995 committed:
Apply suggestions from code review

Co-authored-by: Bonrob2 <59453630+Bonrob2@users.noreply.github.com>

Commit:
ALEC-228: Quick Start guide

- Update quick-start/situations.adoc

GeraMantecon reviewed (approved):
Looks good to me!

BenjaminJ reviewed (approved):
LGTM

Bonrob2 reviewed (approved):
LGTM! Note that it would be better if the situation timeline graphic showed some activity, but let's capture that in a future Jira.
